text/0000-edition-2021.md

```diff
@@ -0,0 +1,238 @@
+
+Key points include:
+
+* Editions are used to introduce changes into the language that would otherwise have the potential to break existing code, such as the introduction of a new keyword.
+* Editions are never allowed to split the ecosystem. We only permit changes that still allow crates in different editions to interoperate.
+* Editions are named after the year in which they occur (e.g., Rust 2015, Rust 2018, Rust 2021).
+* When we release a new edition, we also release tooling to automate the migration of crates. Some manual work may be required but that should be uncommon.
+* The nightly toolchain offers "preview" access to upcoming editions, so that we can land work that targets future editions at any time.
+* We maintain an Edition Migration Guide that offers guidance on how to migrate to the next edition.
+* Whenever possible, new features should be made to work across all editions.
+
+This RFC is meant to establish the high-level purpose of an edition and to describe how the RFC feels to end users. It intentionally avoids detailed policy discussions, which are deferred to the appropriate subteams (compiler, lang, dev-tools, etc) to figure out.
+
+[RFC 2052]: https://github.com/rust-lang/rfcs/blob/master/text/2052-epochs.md
+
+# Motivation
+[motivation]: #motivation
+
+The plan for editions was laid out in [RFC 2052] and Rust had its first edition in 2018. This effort was in many ways a success but also resulted in a lot of stress on the Rust organization as a whole. This RFC proposes a different model for the 2021 Edition. Depending on our experience, we may opt to continue with this model in the future, or we may wish to make changes for future editions.
+
+## Editions enable "stability without stagnation"
+
+The release of Rust 1.0 established ["stability without stagnation"](https://blog.rust-lang.org/2014/10/30/Stability.html) as a core Rust deliverable. Ever since the 1.0 release, the rule for Rust has been that once a feature has been released on stable, we are committed to supporting that feature for all future releases. 
+
+There are times, however, when it is useful to be able to make small changes in the surface syntax of Rust that would not otherwise be backwards compatible. The most obvious example is introducing a new keyword, which would invalidate existing names for variables and so forth. Even when such changes do not "feel" backwards incompatible, they still have the potential to break existing code. If we were to make such changes, people would quickly find that existing programs stopped compiling.
+
+**Editions** are the mechanism we use to square this circle. When we wish to release a feature that would otherwise be backwards incompatible, we do so as part of a new **Rust edition**. Editions are opt-in, and so existing crates do not see these changes until they explicitly migrate over to the new edition. New creates created by cargo always default to the most recent edition.
```

Comment:
```suggestion
**Editions** are the mechanism we use to square this circle. When we wish to release a feature that would otherwise be backwards incompatible, we do so as part of a new **Rust edition**. Editions are opt-in, and so existing crates do not see these changes until they explicitly migrate over to the new edition. New crates created by cargo always default to the most recent edition.
```